### PR TITLE
Add single grid channel capability and test for C-grid

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Add single grid channel capability and test for C-grid
- [X] Developer(s): 
    JFLemieux73, apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    Full test suite run on cheyenne, all results bit-for-bit except gx3/gx1 cases on C and CD grid, https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#fd67ed2615ad77ac7221ae9f766b69cd2172c317.  QC tests on gx1 C grid passes.  This modification allows for transport in single grid channels.  Without it, there can be velocity but no transport.  kmtislands doesn't change results in these tests vs the baseline because transport is off in those tests.
- How much do the PR code changes differ from the unmodified code? 
    - [ ] bit for bit
    - [ ] different at roundoff level
    - [X] more substantial for C and CD grid 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [X] Yes
    - [ ] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [X] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [X] No 
- [X] Please provide any additional information or relevant details below:

- Adds modification to locate_triangles in remap advection to support transport in single grid cell channels and in cells that are closed on three sides.
- Adds single change east and north configurations and tests
- Adds tx1 tripole C and CD grid tests

```
INFO:__main__:Running QC test on the following directories:
INFO:__main__:  /glade/scratch/tcraig/CICE_RUNS/cheyenne_intel_smoke_gx1_36x1_gridc_medium_qc.qc_base
INFO:__main__:  /glade/scratch/tcraig/CICE_RUNS/cheyenne_intel_smoke_gx1_36x1_gridc_medium_qc.qc_test
INFO:__main__:Number of files: 1825
INFO:__main__:2 Stage Test Passed
INFO:__main__:Quadratic Skill Test Passed for Northern Hemisphere
INFO:__main__:Southern Hemisphere data is bit-for-bit
INFO:__main__:Creating map of the data (ice_thickness_cheyenne_intel_smoke_gx1_36x1_gridc_medium_qc.qc_base.png)
INFO:__main__:Creating map of the data (ice_thickness_cheyenne_intel_smoke_gx1_36x1_gridc_medium_qc.qc_te.png)
INFO:__main__:Creating map of the data (ice_thickness_cheyenne_intel_smoke_gx1_36x1_gridc_medium_qc.qc_base_minus_cheyenne_intel_smoke_gx1_36x1_gridc_medium_qc.qc_te.png)
INFO:__main__:
INFO:__main__:Quality Control Test PASSED

```

### CODE REVIEW

**Comment by apcraig** [comment]
This should be ready to go, but we may delay merging to main until the E3SM Icepack sync is complete.

**Comment by apcraig** [comment]
Actually, I think this is ready for review and merging.  We just need to hold off on commits to Icepack for a bit.

**Comment by JFLemieux73** [comment]
This is good. It makes sense that it is identical in the southern hemisphere. All the one grid cell wide 'features' are in the CAA or Hudson Bay.

File: cicecore/cicedyn/infrastructure/ice_grid.F90
Comment (eclare108213) [inline cicecore/cicedyn/infrastructure/ice_grid.F90:1494]: what happens if ny_global is odd?

Comment (apcraig) [reply cicecore/cicedyn/infrastructure/ice_grid.F90:1494]: It doesn't matter, it'll just round down.  ny_global/2 is always a well defined integer value.  The channel will be created at j=ny_global/2, whether ny_global is odd or even.  We could have chosen j=2 or j=ny_global/4 or anything really would work.  ny_global/2 is probably safest.  If ny_global is a relatively small integer, it'll create the channel away from the edge as much as possible.  I guess we run into a problem if ny_global is less than 4, but that's not a particularly useful grid, and I don't think we need to worry about that.

**Review by JFLemieux73 — COMMENTED** [review]
Hi @apcraig,

I added these temporary comments just to remind me to think about this again...I think the code is fine and these comments could be removed. Thanks!

File: cicecore/cicedyn/dynamics/ice_transport_remap.F90
Comment (JFLemieux73) [inline cicecore/cicedyn/dynamics/ice_transport_remap.F90]: I mean here and the other places with temp comment. 

We should keep the comment:
! 1 grid-cell wide channel: dpx,y = 0, edgearea /= 0

**Comment by apcraig** [comment]
We could merge this before weekend testing if someone wants to review and nobody thinks we should wait...

**Review by eclare108213 — APPROVED** [review]
Looks great.

**Pull request merged**
